superdesk/publish/formatters/ninjs_newsroom_formatter.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+from .ninjs_formatter import NINJSFormatter
+import superdesk
+
+
+class NewsroomNinjsFormatter(NINJSFormatter):
+    def __init__(self):
+        self.format_type = 'newsroom ninjs'
+        self.can_preview = False
+        self.can_export = False
+
+    def _format_products(self, article):
+        """
+        Return a list of API product id's that the article matches.
+
+        :param article:
+        :return:
+        """
+        result = superdesk.get_resource_service('product_tests').test_products(article, lookup={'product_type': 'api'})
+        return [p['product_id'] for p in result if p.get('matched', False)]
```

Comment:
it might be useful to include names, so there will be labels for ids if needed.
what about products having name and code like other similar fields?

Reply:
@petrjasek We discussed it, the  thinking was that the size of the ninjs response would potentially  be large if the names were included. 

I think the customer admin permissioning will need to retrieve a complete list of id's and names, I don't think we'd worked out how that would work. 

I guess I can include the names, we can always remove them if it is a problenm

Reply:
I see, using names from items would not let you setup some products in newsroom upfront without having items from superdesk using it, not sure if this could be a problem, but if not lets add the names